source/validate_id.cpp

```diff
@@ -1344,6 +1344,28 @@ bool idUsage::isValid<SpvOpInBoundsAccessChain>(
   return isValid<SpvOpAccessChain>(inst, opcodeEntry);
 }
 
+template <>
+bool idUsage::isValid<SpvOpPtrAccessChain>(const spv_instruction_t* inst,
+                                           const spv_opcode_desc opcodeEntry) {
+  // OpPtrAccessChain's validation rules are similar to OpAccessChain, with one
+  // difference: word 4 must be id of an integer (Element <id>).
+  // The grammar guarantees that there are at least 5 words in the instruction
+  // (i.e. if there are fewer than 5 words, the SPIR-V code will not compile.)
+  int elem_index = 4;
+  // We can remove the Element <id> from the instruction words, and simply call
+  // the validation code of OpAccessChain.
+  spv_instruction_t new_inst = *inst;
+  new_inst.words.erase(new_inst.words.begin() + elem_index);
```

Comment:
At first I thought it was dodgy to reuse validation for a different instruction.  But it's ok because we're only doing validation of Ids in this class.  The instruction word does not matter.